src/FluentValidation/Validators/RangeValidator.cs

```diff
@@ -52,9 +52,9 @@ public abstract class RangeValidator<T, TProperty> : PropertyValidator<T, TPrope
 		protected int Compare(TProperty a, TProperty b) {
 
 			// Use explicitComparer first
-			if (explicitComparer != null)
-				return explicitComparer.Compare(a, b);
-			else if (a is IComparable comparableA)
+			if (_explicitComparer != null)
+				return _explicitComparer.Compare(a, b);
+			else if (a is IComparable<TProperty> comparableA)

```

Comment:
This won't avoid boxing if `a` is a value type. This might seem pedantic but it does make a difference to performance under large workloads, and as part of the 10.0 release I'm trying to remove as much boxing as possible. 
I think having a comparer implementation as in my other comment would be better. 

If you don't have the availability to do this then let me know and I'll pick it up when I get some time.